File: Linked Lists/removeKFromList.py
# Singly-linked lists are already defined with this interface:
# class ListNode(object):
#   def __init__(self, x):
#     self.value = x
#     self.next = None
#
def solution(l, k):
    arr = l
    newList = ListNode(0)
    newList.next = l
    prev = newList
    while(arr):
        if(arr.value == k):
            prev.next = arr.next
        else:
            prev = arr    
            
        arr = arr.next
        
    return newList.next

File: Linked Lists/removeKFromList.php
// Singly-linked lists are already defined with this interface:
// class ListNode {
//   public $value;
//   public $next;
//
//   public function __construct($x) {
//     $this->value = $x;
//     $this->next = null;
//   }
// }
//

function solution($l, $k) {
    $newList = new ListNode(0);
    $newList->next = $l;
    $prev = $newList;
    while($l){
        if($l->value == $k){
            $prev->next=$l->next;
        }else{
            $prev = $l;
        }
        $l = $l->next;
    }
    return $newList->next;
    
}
